src/xcsync/ContinuousSyncContext.cs

```diff
@@ -49,27 +59,37 @@ async void ClrFileRenamed (string oldPath, string newPath)
 
 		async void XcodeFileRenamed (string oldPath, string newPath)
 		{
+			if (token.IsCancellationRequested)
+				return;
+
 			Logger.Debug ($"Xcode Project file {oldPath} renamed to {newPath}");
 			await Hub.PublishAsync (ChangeChannel, new ChangeMessage (Guid.NewGuid ().ToString (), newPath, SyncDirection.FromXcode, clrChanges, xcodeChanges));
 		}
 
 		xcodeChanges.OnFileRenamed = XcodeFileRenamed;
 
 		clrChanges.OnError = ex => {
+			if (token.IsCancellationRequested)
+				return;
+
 			Logger.Error (ex, $"Error:{ex.Message} in CLR Project file change monitor");
 		};
 
 		xcodeChanges.OnError = ex => {
+			if (token.IsCancellationRequested)
```

Comment:
just curious, what is the current behavior that provoked the need for the extra cancellation checks?